### PR TITLE
Assignment 5 - Ian Kaiserman

### PR DESCRIPTION
**Pull request by Afterlyte**
Paired with @TannerStephenson

[Website link](https://kind-stone-01ab5711e.3.azurestaticapps.net/)

Still in progress: 
 - Polishing front-end of Word Editor page
 - Fixing pagination issues when searching
 - Ironing out issues with Identity system and getting it to work with seeded credentials (JWT tokens are confusing)

Feedback very much appreciated on the Auth stuff, especially with the token system. A test endpoint is written but isn't working correctly so we're investigating what's causing the issues. We'll update the PR with test seeded credentials once we get it working.

### CODE REVIEW

**Comment by roydunn19** [comment]
Website currently doesn't support deletion of words

**Comment by roydunn19** [comment]
list of words doesn't have option to paginate

**Comment by roydunn19** [comment]
changing common word isnt supported in word editor

File: Wordle.Api/Wordle.Api/Controllers/WordController.cs [hunk 1]
```diff
@@ -52,5 +52,23 @@ public async Task<IEnumerable<WordOfTheDayStatsDto>> GetWordOfTheDayStats(DateTi
...
         }
+
+        [HttpGet("WordList")]
+        public async Task<IEnumerable<Word>> GetWordList(int pageNumber = 1, int pageLength = 20, string? searchString = null)
+        {
+            return await _wordService.GetWordList(pageNumber, pageLength, searchString);
+        }
+
+        [HttpGet("DeleteWord")]
+        public async Task<bool> DeleteWord(int wordId)
```
Comment (roydunn19) [inline Wordle.Api/Wordle.Api/Controllers/WordController.cs:63]: why return bool here?

**Comment by roydunn19** [comment]
layout of word editor is a bit strange with it taking 100% of screen on the top. just my opinion though

**Comment by arose-13** [comment]
Will review.

**Review by arose-13 — COMMENTED** [review]
* Create a page named "WordEditor" that allows words to be edited ✅
  * Support deleting a word ❌ (endpoint is present, but couldn't find how to do this)
  * Support adding a new word ✅
  * Editing words is not supported ✅
  * Duplicate words should not be allowed ✅
  * Support changing the common word boolean flag ✅
  * Words should be sorted alphabetically ✅
* Provide a text box for searching words AS YOU TYPE which supports searching for words starting with letters specified ❌
* Provide a menu item for navigating to the WordEditor page ✅
  * Anyone can look at the word list ✅
  * Word list is paginated (10-100 per page) ❌
* Any logged in user can change the common word flag ✅
  * There is a login page/component/popup/etc for users to sign in ✅
* Only users over 21 years of age (based on birthday) with a claim of MasterOfTheUniverse can add and remove words ❌
* Users can be set up at start up in the API project as a seed (apriori), there does not need to be a sign up mechanism ✅

There are a lot of 404 in the console, and I couldn't see any words on your WordEditor page. Also, based on your endpoints that you have written, it appears that accessing your `WordController` doesn't require any authorization. I also couldn't find a way to make an account, so I'm not sure if being a guest broke some things or not.
Screenshot (448)
Screenshot (450)

File: Wordle.Api/Wordle.Api/Controllers/WordController.cs [hunk 2]
```diff
@@ -52,5 +54,25 @@ public async Task<IEnumerable<WordOfTheDayStatsDto>> GetWordOfTheDayStats(DateTi
...
+
+        [HttpGet("DeleteWord")]
+        [Authorize(Policy = Policies.CanCreateDeleteWords)]
+        public async Task<bool> DeleteWord(int wordId)
+        {
+            return await _wordService.DeleteWord(wordId);
+        }
+
+        [HttpGet("ChangeCommonFlag")]
+        [Authorize(Policy = Policies.CanCreateDeleteWords)]
```
Comment (meghanmae) [inline Wordle.Api/Wordle.Api/Controllers/WordController.cs:72]: Anyone logged in should be able to update the common word flag, so it should be a different policy

File: Wordle.Api/Wordle.Api/Controllers/WordController.cs [hunk 3]
```diff
@@ -52,5 +54,25 @@ public async Task<IEnumerable<WordOfTheDayStatsDto>> GetWordOfTheDayStats(DateTi
...
+
+        [HttpGet("WordList")]
+        public async Task<IEnumerable<Word>> GetWordList(int pageNumber = 1, int pageLength = 20, string? searchString = null)
+        {
+            return await _wordService.GetWordList(pageNumber, pageLength, searchString);
+        }
+
+        [HttpGet("DeleteWord")]
+        [Authorize(Policy = Policies.CanCreateDeleteWords)]
+        public async Task<bool> DeleteWord(int wordId)
```
Comment (meghanmae) [inline Wordle.Api/Wordle.Api/Controllers/WordController.cs:66]: Your add word function also needs some protection

**Pull request closed without merging**